### PR TITLE
SB-29068 modified certificate download api wrto sunbird-rc integration

### PR DESCRIPTION
**Pull request by Hari-stackroute**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-29068" title="SB-29068" target="_blank">TaskSB-29068</a>  Certificate Registry Integration with RC for download and search certificate
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Java-11, play2-2.7.2
* Hardware versions: 2 CPU/ 4GB RAM

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: all-actors/src/main/java/org/sunbird/RegistryCredential.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * this class will help in getting the env values for calling the cert service
+ * @author anmolgupta
+ */
+public class RegistryCredential {
+
+    public static final String REGISTRY_CREDENTIAL_SERVICE_BASE_URL= "registry_credential_service_base_url";
+    private static Logger logger= LoggerFactory.getLogger(RegistryCredential.class);
+    private  static final String SERVICE_BASE_URL=getPropsFromEnvs(REGISTRY_CREDENTIAL_SERVICE_BASE_URL);
+    private static final String DOWNLOAD_URI= "/rc/certificate/v1/download";
```
Comment (reshmi-nair) [inline all-actors/src/main/java/org/sunbird/RegistryCredential.java]: Change this url to service url and configure schema name as env variable

File: all-actors/src/main/java/org/sunbird/RegistryCredential.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    private static String getPropsFromEnvs(String props){
+        String propValue=System.getenv(props);
+        return propValue;
+    }
+
+
+    public static String getEsSearchUri(){
+        String esApi=String.format("http://%s:9200/%s/search",getPropsFromEnvs(SUNBIRD_ES_IP).split(",")[0],"   rc/certificate/v1");
```
Comment (reshmi-nair) [inline all-actors/src/main/java/org/sunbird/RegistryCredential.java]: change this to RC search url

File: all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java [hunk 3]
```diff
@@ -246,8 +247,25 @@ public Response downloadV2(Request request) throws BaseException {
...
             }
         } else {
-            throw new BaseException(IResponseMessage.RESOURCE_NOT_FOUND, localizer.getMessage(IResponseMessage.RESOURCE_NOT_FOUND, null), ResponseCode.RESOURCE_NOT_FOUND.getCode());
+            Map<String, String> headerMap = new HashMap<>();
+            headerMap.put("Accept", "application/vc+ld+json");
+            String rcTemplateApi = RegistryCredential.getSERVICE_BASE_URL().concat(RegistryCredential.getDOWNLOAD_URI())+"/"+certId;
+            Future<HttpResponse<JsonNode>> rcResponseFuture=CertificateUtil.makeAsyncGetCall(rcTemplateApi,headerMap);
+            HttpResponse<JsonNode> rcJsonResponse = rcResponseFuture.get();
+            if (rcJsonResponse != null && rcJsonResponse.getStatus() == HttpStatus.SC_OK) {
+                String templateUrl = rcJsonResponse.getBody().getObject().getJSONObject(JsonKeys.RESULT).getString("templateurl");
```
Comment (reshmi-nair) [inline all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java]: change to templateUrl and define jsonkey

File: all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java [hunk 4]
```diff
@@ -246,8 +247,25 @@ public Response downloadV2(Request request) throws BaseException {
...
+                String rcDownloadApi = RegistryCredential.getSERVICE_BASE_URL().concat(RegistryCredential.getDOWNLOAD_URI())+"/"+certId;
+                headerMap.put("Accept", "image/svg+xml");
+                headerMap.put("templateurl", templateUrl);
```
Comment (reshmi-nair) [inline all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java]: jsonkey needed

File: all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java [hunk 5]
```diff
@@ -246,8 +247,25 @@ public Response downloadV2(Request request) throws BaseException {
...
+                Future<HttpResponse<JsonNode>> rcDownloadResFuture=CertificateUtil.makeAsyncGetCall(rcDownloadApi,headerMap);
+                HttpResponse<JsonNode> rcDownloadJsonResponse = rcDownloadResFuture.get();
+                printUri =  rcDownloadJsonResponse.getBody().getObject().getJSONObject(JsonKeys.RESULT).getString(JsonKeys.PRINT_URI);
```
Comment (reshmi-nair) [inline all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java]: We will be getting svg image String. Check the current download folow from UI. set the string to print uri

File: all-actors/src/main/java/org/sunbird/RegistryCredential.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * this class will help in getting the env values for calling the cert service
+ * @author anmolgupta
+ */
+public class RegistryCredential {
+
+    public static final String REGISTRY_CREDENTIAL_SERVICE_BASE_URL= "registry_credential_service_base_url";
+    private static Logger logger= LoggerFactory.getLogger(RegistryCredential.class);
+    private  static final String SERVICE_BASE_URL = getPropsFromEnvs(REGISTRY_CREDENTIAL_SERVICE_BASE_URL);
+    private  static final String CERTIFICATE_TYPE = getPropsFromEnvs("certificate_type");
```
Comment (Hari-stackroute) [inline all-actors/src/main/java/org/sunbird/RegistryCredential.java]: change it to RC_ENTITY

**Review by reshmi-nair — CHANGES_REQUESTED** [review]

**Review by reshmi-nair — APPROVED** [review]

**Review by reshmi-nair — APPROVED** [review]

**Review by reshmi-nair — APPROVED** [review]

**Pull request merged**